docs/shone/examples/chemistry.rst

```diff
@@ -0,0 +1,309 @@
+
+.. code-block:: python
+
+    import numpy as np
+
+    pressure = np.geomspace(1e-6, 10, 15)  # [bar]
+    temperature = 2300 * (pressure / 0.1) ** 0.1  # [K]
+
+.. plot::
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    pressure = np.geomspace(1e-6, 10, 15)  # [bar]
+    temperature = 2300 * (np.geomspace(1e-6, 10, pressure.size) / 0.1) ** 0.1  # [K]
+
+    ax = plt.gca()
+    ax.semilogy(temperature, pressure)
+    ax.invert_yaxis()
+    ax.set(
+        xlabel='Temperature [K]',
+        ylabel='Pressure [bar]'
+    )
+
+Radiative transfer calculations require the mixing ratio of each atmospheric species, which are
+functions of temperature, pressure, atmospheric metallicity, and C/O ratio. We specify these
+parameters like so:
+
+.. code-block:: python
+
+    from shone.chemistry import FastchemWrapper
+
+    chem = FastchemWrapper(
+        temperature, pressure,
+        # these are in linear (not log) units:
+        metallicity=1,  # M/H
+        c_to_o_ratio=1  # C/O
+    )
+
+The wrapper can enumerate the names, symbols, weights, and indices for each
+species with `~shone.chemistry.FastchemWrapper.get_species`:
+
+.. code-block:: python
+
+    species_table = chem.get_species()
+    print(species_table[:3])  # first three species
+
+Returns a table like this:
+
+.. raw:: html
+
+    <br />
+    <table style="width:75%">
+      <thead>
+        <tr style="text-align: left;">
+          <th></th>
+          <th>index</th>
+          <th>name</th>
+          <th>weight</th>
+          <th>type</th>
+        </tr>
+        <tr>
+          <th>symbol</th>
+          <th></th>
+          <th></th>
+          <th></th>
+          <th></th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <th>e-</th>
+          <td>0</td>
+          <td>Electron</td>
+          <td>0.000549</td>
+          <td>element</td>
+        </tr>
+        <tr>
+          <th>Al</th>
+          <td>1</td>
+          <td>Aluminium</td>
+          <td>26.981538</td>
+          <td>element</td>
+        </tr>
+        <tr>
+          <th>Ar</th>
+          <td>2</td>
+          <td>Argon</td>
+          <td>39.948000</td>
+          <td>element</td>
+        </tr>
+      </tbody>
+    </table>
+    <br />
+    <br />
+
+Volume mixing ratio
++++++++++++++++++++
+
+We compute the matrix of volume mixing ratios for all species with
+`~shone.chemistry.FastchemWrapper.vmr`, and index it for a particular
+species:
+
+.. code-block:: python
+
+    # lookup the column index for O2 in the fastchem VMR matrix:
+    idx = species_table.loc['O2']['index']
+    vmr_O2 = vmr[: idx]
+
+We can plot the VMRs of several species as a function of pressure like so:
+
+.. code-block:: python
+
+    import matplotlib.pyplot as plt
+
+    species = ['H2O1', 'O1Ti1', 'e-']
+    indices = species_table.loc[species]['index']
+    names = species_table.loc[species]['name']
+
+    ax = plt.gca()
+    ax.loglog(vmr[:, indices], pressure, label=names)
```

Comment:
And: `ax.loglog(chem.vmr()[:, indices], pressure, label=names)`

Reply:
Thanks!